core/templates/dev/head/domain/question/QuestionObjectFactorySpec.ts

```diff
@@ -166,8 +166,12 @@ describe('Question object factory', function() {
       'id', 'name', 'notes', 'feedback');
     var misconception2 = MisconceptionObjectFactory.create(
       'id_2', 'name_2', 'notes', 'feedback');
+    var misconceptionsDict = {
+      skillId1: [misconception1],
+      skillId2: [misconception2]
```

Comment:
Add a case with multiple misconceptions for a given skill, make sure all possible errors in that case are caught

Reply:
Done.